zerver/lib/streams.py

```diff
@@ -918,20 +917,7 @@ def public_stream_user_ids(stream: Stream) -> set[int]:
     guest_subscriptions_ids = {
         sub["user_profile_id"] for sub in guest_subscriptions.values("user_profile_id")
     }
-    can_add_subscribers_group_user_ids = set(
-        get_recursive_group_members(stream.can_add_subscribers_group_id)
-        .exclude(
-            # allow_everyone_group=False is false for can_add_subscribers_group,
-            # so guest users cannot exercise this permission.
-            role=UserProfile.ROLE_GUEST
-        )
-        .values_list("id", flat=True)
```

Comment:
We would need this code later for `can_subscribe_group` when `allow_everyone_group` will be `true`. But currently all non-guests are already included.

Reply:
Yeah, it's reasonable to remove this, even if we'll be adding it back fairly soon.